csrc/device_lower/validation.cpp

```diff
@@ -206,6 +206,191 @@ void validateCpAsyncBulk(const std::vector<TensorView*>& tvs) {
+
+    // The input will be initialized for this argsort. To avoid any
+    // potential conflict of initialization, require the input to be
+    // exclusively used by the argsort
+    NVF_ERROR_EQ(inp_tv->uses().size(), 1);
+
+    IterDomain* grouped_id = nullptr;
+    for (const auto& loop_id : out_tv->getLoopDomain()) {
+      if (loop_id->getParallelType() == ParallelType::Group) {
+        NVF_ERROR(grouped_id == nullptr, "Multiple IDs found to be grouped");
+        grouped_id = loop_id;
+      }
+    }
+
+    if (grouped_id != nullptr) {
+      // If it's grouped, it must correspond to the innermost subregion
+      // of the logical argsort ID
+      IterDomain* argsort_logical_id =
+          out_tv->getLogicalDomain().at(aop->dim());
+      NVF_ERROR(
+          isInnermost(argsort_logical_id, grouped_id),
+          "Invalid ID to group: ",
+          grouped_id->toString(),
+          " of ",
+          aop->toString());
+
+      // The output is allocated on registers, so the allocation
+      // domain should be just the same as the loop domain. The
+      // grouped ID must have the unit stride, which means all of the
+      // inner IDs must not contribute to the allocation
+      validateUnitStride(out_tv, out_tv->getLoopDomain(), grouped_id);
+
+      // All of the inputs per thread must be provided to the device
+      // function as a contiguous chunk of memory where each element
+      // has a unit stride within its allocated buffer. More
+      // concretely, the input would be passed to the device function
+      // as follows:
+      //
+      //  // Each thread computes 8 argsort elements
+      //  float T1[8];
+      //  ...
+      //  blockArgsort(..., T1, ...);
+      //
+      // Here, the input is required to have a loop ID that is exactly
+      // mapped with the grouped loop ID of the output, and that
+      // producer loop ID is required to use the Serial parallel
+      // type. Furthermore, like the output, none of the inner loop
+      // IDs must not contribute to the allocation of the input so that the
+      // grouped ID has a unit stride.
+      //
+      // The requirement of the input being transformed exactly in the
+      // same as the output is a sufficient but not required
+      // condition. It could be relaxed if necessary.
+      const auto& c2p_replay = BestEffortReplay::replayPasC(
+          inp_tv, out_tv, -1, PairwiseLogicalDomainMap(inp_tv, out_tv));
+      auto producer_grouped_id_it = c2p_replay.getReplay().find(grouped_id);
+      NVF_ERROR(
+          producer_grouped_id_it != c2p_replay.getReplay().end(),
+          "No corresponding producer ID for the grouped consumer ID found: ",
+          grouped_id->toString());
+      auto producer_grouped_id = producer_grouped_id_it->second;
+
+      auto loop_id_it =
+          std::ranges::find(inp_tv->getLoopDomain(), producer_grouped_id);
+      NVF_ERROR(
+          loop_id_it != inp_tv->getLoopDomain().end(),
```

Comment:
nitpick:
```suggestion
      NVF_ERROR(
          std::ranges::contains(inp_tv->getLoopDomain(), producer_grouped_id),
```

Reply:
Actually, that seems like a C++23 feature.